$OpenBSD: patch-dyncallback_dyncall_callback_ppc32_S,v 1.2 2020/08/02 14:06:13 cwen Exp $

Hunk #1: 'machine ppc' is not supported by clang's integrated assembler
          on powerpc

https://dyncall.org/pub/dyncall/dyncall/rev/d982a00c2177

Index: dyncallback/dyncall_callback_ppc32.S
--- dyncallback/dyncall_callback_ppc32.S.orig
+++ dyncallback/dyncall_callback_ppc32.S
@@ -25,7 +25,6 @@
 
 #include "../portasm/portasm-ppc.S"
 
-.machine ppc
 .text
 .align 2
 
@@ -123,9 +122,9 @@ Input:
 	mtctr   r12
 	bctrl
 	/* check result type */
-	cmpi    cr0, r3, 0x66 /* 'f */
+	cmpi    cr0, 0, r3, 0x66 /* 'f */
 	beq .f32
-	cmpi    cr0, r3, 0x64 /* 'd */
+	cmpi    cr0, 0, r3, 0x64 /* 'd */
 	beq .f64
 .i64:
 	lwz     r3, SP_RESULT     (r1)
